Community/Tdarr_Plugin_vdka_Tiered_CPU_CRF_Based_Configurable.js

```diff
@@ -0,0 +1,200 @@
+      //mitigate TrueHD audio causing Too many packets error
+      try {
+        if (
+          file.ffProbeData.streams[i].codec_name.toLowerCase() == 'truehd' ||
+          (file.ffProbeData.streams[i].codec_name.toLowerCase() == 'dts' &&
+            file.ffProbeData.streams[i].profile.toLowerCase() == 'dts-hd ma') ||
+          (file.ffProbeData.streams[i].codec_name.toLowerCase() == 'aac' &&
+            file.ffProbeData.streams[i].sample_rate.toLowerCase() == '44100' &&
+            file.ffProbeData.streams[i].codec_type.toLowerCase() == 'audio')
+        ) {
+          maxmux = ` -max_muxing_queue_size 9999`
+        }
+      } catch (err) {}
+      //mitigate errors due to embeded pictures
+      try {
+        if (
+          (file.ffProbeData.streams[i].codec_name.toLowerCase() == 'png' ||
+            file.ffProbeData.streams[i].codec_name.toLowerCase() == 'bmp' ||
+            file.ffProbeData.streams[i].codec_name.toLowerCase() == 'mjpeg') &&
+          file.ffProbeData.streams[i].codec_type.toLowerCase() == 'video'
+        ) {
+          map = `-map 0:v:0 -map 0:a -map 0:s?`
+        }
+      } catch (err) {}
+    }
+    //file will be encoded if the resolution is 480p or 576p
+    //codec will be checked so it can be transcoded correctly
+    if (file.video_resolution === '480p' || file.video_resolution === '576p') {
+      CRFinuse = `${inputs.sdCRF}`
+      response.preset += `,${map} -dn -c:v libx265 -preset ${ffmpeg_preset} -x265-params crf=${inputs.sdCRF}:bframes=${inputs.bframe}:rc-lookahead=32:ref=6:b-intra=1:aq-mode=3 -a53cc 0 -c:a copy ${subcli}${maxmux}`
+      transcode = 1
+    }
+  
+    //file will be encoded if the resolution is 720p
+    //codec will be checked so it can be transcoded correctly
+    if (file.video_resolution === '720p') {
+      CRFinuse = `${inputs.hdCRF}`
+      response.preset += `,${map} -dn -c:v libx265 -preset ${ffmpeg_preset} -x265-params crf=${inputs.hdCRF}:bframes=${inputs.bframe}:rc-lookahead=32:ref=6:b-intra=1:aq-mode=3 -a53cc 0 -c:a copy ${subcli}${maxmux}`
+      transcode = 1
+    }
+    //file will be encoded if the resolution is 1080p
+    //codec will be checked so it can be transcoded correctly
+    if (file.video_resolution === '1080p') {
+      CRFinuse = `${inputs.fullhdCRF}`
+      response.preset += `,${map} -dn -c:v libx265 -preset ${ffmpeg_preset} -x265-params crf=${inputs.fullhdCRF}:bframes=${inputs.bframe}:rc-lookahead=32:ref=6:b-intra=1:aq-mode=3 -a53cc 0 -c:a copy ${subcli}${maxmux}`
+      transcode = 1
+    }
+    //file will be encoded if the resolution is 4K
+    //codec will be checked so it can be transcoded correctly
+    if (file.video_resolution === '4KUHD') {
+      CRFinuse = `${inputs.uhdCRF}`
+      response.preset += `,${map} -dn -c:v libx265 -preset ${ffmpeg_preset} -x265-params crf=${inputs.uhdCRF}:bframes=${inputs.bframe}:rc-lookahead=32:ref=6:b-intra=1:aq-mode=3 -a53cc 0 -c:a copy ${subcli}${maxmux}`
+      transcode = 1
+    }
```

Comment:
Perhaps you could clean this up a bit by creating a switch that sets `CRFinuse`, then using that for a single `response.preset` instead of these 4 if statements. You could use the `default` case of the switch to abort the plugin.

Reply:
This has been done